### ISSUE TITLE
NaN loss in deepvoice3 training

### ISSUE THREAD

**Issue by TheFlash10** [PaddlePaddle/Parakeet#13]
Hey, i am currently training a deepvoice3 model on a custom dataset, i am facing this error

2020-06-04 12:01:42,075-WARNING: NaN or Inf found in input tensor.
2020-06-04 12:01:42,075-WARNING: NaN or Inf found in input tensor.
global_step: 20999      loss: nan
38it [01:14,  2.21s/it]2020-06-04 12:01:43,406-WARNING: NaN or Inf found in input tensor.
2020-06-04 12:01:43,406-WARNING: NaN or Inf found in input tensor.
2020-06-04 12:01:43,407-WARNING: NaN or Inf found in input tensor.
2020-06-04 12:01:43,407-WARNING: NaN or Inf found in input tensor.
2020-06-04 12:01:43,407-WARNING: NaN or Inf found in input tensor.
2020-06-04 12:01:43,407-WARNING: NaN or Inf found in input tensor.
global_step: 21000      loss: nan
39it [01:17,  1.95s/it]2020-06-04 12:01:45,869-WARNING: NaN or Inf found in input tensor.
2020-06-04 12:01:45,870-WARNING: NaN or Inf found in input tensor.
2020-06-04 12:01:45,870-WARNING: NaN or Inf found in input tensor.
2020-06-04 12:01:45,871-WARNING: NaN or Inf found in input tensor.
2020-06-04 12:01:45,872-WARNING: NaN or Inf found in input tensor.
2020-06-04 12:01:45,872-WARNING: NaN or Inf found in input tensor.

The loss comes out to be 'nan'. When i rerun and resume from the latest checkpoint, it works perfectly. Still sometimes the error pops up.

My understanding is that probably the learning rate is high, resulting in 'nan' loss. I have used the default config file and just the sampling rate is different.

Can you help me by suggesting something to solve this issue ?

Thanks

**Comment by iclementine** [comment]
I think you can first try resampling the audio to 22050Hz when loading the audio with librosa, then we can check whether the code has a bug dealing with other sample rates. Or may be it is caused by other factors.

```
librosa.core.load(path, sr=22050, mono=True, offset=0.0, duration=None, dtype=<class 'numpy.float32'>, res_type='kaiser_best')
```

https://github.com/PaddlePaddle/Parakeet/blob/b7c584e2f735a3ab2b2bcaf1ecd7b636227c5646/examples/deepvoice3/data.py#L97

**Comment by aayushkubb** [comment]
Hey ,

What effect does resampling have on training? I believe the more the sampling rate the better features we have and less loss of information.

**Comment by TheFlash10** [comment]
> I think you can first try resampling the audio to 22050Hz when loading the audio with librosa, then we can check whether the code has a bug dealing with other sample rates. Or may be it is caused by other factors.
> 
> ```
> librosa.core.load(path, sr=22050, mono=True, offset=0.0, duration=None, dtype=<class 'numpy.float32'>, res_type='kaiser_best')
> ```
> 
> https://github.com/PaddlePaddle/Parakeet/blob/b7c584e2f735a3ab2b2bcaf1ecd7b636227c5646/examples/deepvoice3/data.py#L97

Hey,
So i tried the training after resampling the audios to 22050Hz, it works perfectly now. Probably it was having some problem dealing with higher sampling rate. What can be the reason for having issue with higher sampling rates ?


Thanks

**Comment by iclementine** [comment]
Higher sampling rate means higher time resolution. Then to extract spectrogram feature from the audio, you may need a larger `n_fft`, `win_length` and `hop_length` to capture the spectrogram features.

You can compute the time corresponding to a `frame` by `n_fft / sample_rate` and the time shift between two adjacent frames by `hop_length / sample_rate` to set proper values for them. Check librosa's documentation for more details.

But how does higher sampling rate cause Nan loss, we are not certain yet.
